* Mon Jul 09 2012 Sérgio Basto <sergio@serjux.com> 
  rfbz #1826, last review of https://bugzilla.rpmfusion.org/attachment.cgi?id=888
  by: Naveed Hasan <naveed@cs.columbia.edu>

* Sat Jun 16 2012 Sérgio Basto <sergio@serjux.com>
  Rebased for 4.1.16 (partial rfbz #1826)


From: Lubomir Rintel <lkundrak@v3.sk>
Date: Mon, 10 May 2010 12:08:47 -0400
Subject: [PATCH] Replace some strings

With ones that (sometimes) make more sense.
---
 src/VBox/Frontends/VirtualBox/src/main.cpp |   20 ++++++++++----------
 src/VBox/Installer/linux/VBox.sh           |   11 ++++++-----
 src/VBox/VMM/VMMR3/VM.cpp                  |    8 ++++----
 3 files changed, 20 insertions(+), 19 deletions(-)


diff -Nurp VirtualBox-4.1.18/src/VBox/Frontends/VirtualBox/src/main.cpp VirtualBox-4.1.18.new2/src/VBox/Frontends/VirtualBox/src/main.cpp
--- VirtualBox-4.1.18/src/VBox/Frontends/VirtualBox/src/main.cpp	2012-06-20 14:16:20.000000000 +0100
+++ VirtualBox-4.1.18.new2/src/VBox/Frontends/VirtualBox/src/main.cpp	2012-07-09 04:57:00.690645723 +0100
@@ -72,13 +72,16 @@ QString g_QStrHintLinuxNoMemory = QAppli
   );
 
 QString g_QStrHintLinuxNoDriver = QApplication::tr(
-  "The VirtualBox Linux kernel driver (vboxdrv) is either not loaded or "
-  "there is a permission problem with /dev/vboxdrv. Please reinstall the kernel "
-  "module by executing<br/><br/>"
-  "  <font color=blue>'/etc/init.d/vboxdrv setup'</font><br/><br/>"
-  "as root. If it is available in your distribution, you should install the "
-  "DKMS package first. This package keeps track of Linux kernel changes and "
-  "recompiles the vboxdrv kernel module if necessary."
+  "The VirtualBox Linux kernel driver (vboxdrv) is probably not loaded."
+  "You may not have kernel driver installed for kernel that is runnig, if so you may do as root:"
+  "  <font color=blue>'yum install kmod-VirtualBox-$(uname -r) kmod-VirtualBox'</font><br/><br/>"
+  "On *not* stable versions of Fedora you need use akmods (because we don't have pre-built kmods) "
+  "or for custom kernels, you may do instead as root:"
+  "  <font color=blue>'yum install akmod-VirtualBox kernel-devel-$(uname -r)'</font>"
+  "  <font color=blue>'akmods'</font><br/><br/>"
+  "If you installed VirtualBox packages and don't want reboot the system, you may need load the "
+  "kernel driver, doing as root:"
+  "  <font color=blue>'systemctl restart systemd-modules-load.service'</font><br/><br/>"
   );
 
 QString g_QStrHintOtherWrongDriverVersion = QApplication::tr(
@@ -92,18 +92,18 @@ QString g_QStrHintLinuxWrongDriverVersio
   "The VirtualBox kernel modules do not match this version of "
   "VirtualBox. The installation of VirtualBox was apparently not "
   "successful. Executing<br/><br/>"
-  "  <font color=blue>'/etc/init.d/vboxdrv setup'</font><br/><br/>"
+  "  <font color=blue>'systemctl restart systemd-modules-load.service'</font> as root<br/><br/>"
   "may correct this. Make sure that you do not mix the "
-  "OSE version and the PUEL version of VirtualBox."
+  "RPM Fusion version and the Oracle version of VirtualBox."
   );
 
 QString g_QStrHintOtherNoDriver = QApplication::tr(
   "Make sure the kernel module has been loaded successfully."
   );
 
-/* I hope this isn't (C), (TM) or (R) Microsoft support ;-) */
+/* We're going to try to give a helping hand */
 QString g_QStrHintReinstall = QApplication::tr(
-  "Please try reinstalling VirtualBox."
+  "Please report the problem to the RPM Fusion Bugzilla."
   );
 
 #if defined(DEBUG) && defined(Q_WS_X11) && defined(RT_OS_LINUX)
diff -Nurp VirtualBox-4.1.18/src/VBox/Installer/linux/VBox.sh VirtualBox-4.1.18.new2/src/VBox/Installer/linux/VBox.sh
--- VirtualBox-4.1.18/src/VBox/Installer/linux/VBox.sh	2012-06-20 14:17:29.000000000 +0100
+++ VirtualBox-4.1.18.new2/src/VBox/Installer/linux/VBox.sh	2012-07-09 05:02:47.158108166 +0100
@@ -32,19 +32,22 @@ elif ! lsmod|grep -q vboxdrv; then
     cat << EOF
 WARNING: The vboxdrv kernel module is not loaded. Either there is no module
          available for the current kernel (`uname -r`) or it failed to
-         load. Please recompile the kernel module and install it by
+         load. Please make sure that you have kmod-VirtualBox for current kernel and load the kernel module by executing
 
-           sudo /etc/init.d/vboxdrv setup
+           'systemctl restart systemd-modules-load.service' (as root)
 
          You will not be able to start VMs until this problem is fixed.
 EOF
 elif [ ! -c /dev/vboxdrv ]; then
     cat << EOF
-WARNING: The character device /dev/vboxdrv does not exist. Try
+WARNING: The character device /dev/vboxdrv does not exist. Try executing
 
-           sudo /etc/init.d/vboxdrv restart
+           '/sbin/udevadm control --reload-rules' (as root)
+           '/sbin/udevadm trigger --subsystem-match=usb --action=add' (as root)
+           '/sbin/udevadm settle' (as root)
 
-         and if that is not successful, try to re-install the package.
+         and if that is not successful, then you're in situation that
+         can not possibly happen and there's nothing you can do about it.
 
          You will not be able to start VMs until this problem is fixed.
 EOF
diff -Nurp VirtualBox-4.1.18/src/VBox/VMM/VMMR3/VM.cpp VirtualBox-4.1.18.new2/src/VBox/VMM/VMMR3/VM.cpp
--- VirtualBox-4.1.18/src/VBox/VMM/VMMR3/VM.cpp	2012-06-20 14:18:34.000000000 +0100
+++ VirtualBox-4.1.18.new2/src/VBox/VMM/VMMR3/VM.cpp	2012-07-09 05:01:48.544521745 +0100
@@ -375,8 +375,8 @@ VMMR3DECL(int)   VMR3Create(uint32_t cCp
 #ifdef RT_OS_LINUX
                     pszError = N_("VirtualBox kernel driver not loaded. The vboxdrv kernel module "
                                   "was either not loaded or /dev/vboxdrv is not set up properly. "
-                                  "Re-setup the kernel module by executing "
-                                  "'/etc/init.d/vboxdrv setup' as root");
+                                  "Load the kernel module by executing "
+                                  "'systemctl restart systemd-modules-load.service' as root");
 #else
                     pszError = N_("VirtualBox kernel driver not loaded");
 #endif
@@ -417,8 +417,8 @@ VMMR3DECL(int)   VMR3Create(uint32_t cCp
 #ifdef RT_OS_LINUX
                     pszError = N_("VirtualBox kernel driver not installed. The vboxdrv kernel module "
                                   "was either not loaded or /dev/vboxdrv was not created for some "
-                                  "reason. Re-setup the kernel module by executing "
-                                  "'/etc/init.d/vboxdrv setup' as root");
+                                  "reason. Install the kernel module by executing "
+                                  "'yum install kmod-VirtualBox-$(uname -r) kmod-VirtualBox' as root");
 #else
                     pszError = N_("VirtualBox kernel driver not installed");
 #endif
